frontend/catalyst/pennylane_extensions.py

```diff
@@ -269,31 +318,125 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    methods = {"fd", "ps", "adj"}
-    if method is None:
-        method = "fd"
-    if method not in methods:
-        raise ValueError(
-            f"Invalid differentiation method '{method}'. "
-            f"Supported methods are: {' '.join(sorted(methods))}"
-        )
-    if method == "fd" and h is None:
-        h = 1e-7
-    if not (h is None or isinstance(h, numbers.Number)):
-        raise ValueError(f"Invalid h value ({h}). None or number was excpected.")
-    if argnum is None:
-        argnum = [0]
-    elif isinstance(argnum, int):
-        argnum = [argnum]
+    return Grad(_ensure_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+
+
+def jvp(f, params, tangents, *, method=None, h=None, argnum=None):
+    """A Jacobian-vector product for PennyLane/Catalyst.
+
+    This function allows the Jacobian-vector Product of a hybrid quantum-classical function to be
+    computed within the compiled program.
+
+    Args:
+        f(DifferentiableLike): Function-like object to calculate JVP for
```

Comment:
```suggestion
        f (Callable): Function-like object to calculate JVP for
```
I believe this is the standard way of documenting this in most python docstrings? E.g., it coincides with what both PL and [jax](https://jax.readthedocs.io/en/latest/_autosummary/jax.grad.html) use.

Reply:
Not every callable object will match here, for example, partial functions will not, so I put a precise type name declared [here](https://github.com/PennyLaneAI/catalyst/pull/98/files#diff-8754bab66a7845aa8e14eb1ab898192548b9ab878d595f57b9c6639930e2c6bfR146).  Consider the following example, that may or may not be an issue. Let me know if you still want to use weaker descriptions.

``` python
import jax.numpy as jnp
from catalyst import qjit, jvp
from numpy.testing import assert_allclose
from functools import partial

def f(x, y):
    return x+y

x = jnp.zeros([4], dtype=float)

@qjit(verbose=True, keep_intermediate=True)
def workflow():
    return jvp(partial(f,y=1), x, x) # Nope

workflow()
```

Reply:
That makes sense!

I'm happy to use `DifferentiableLike` for the type hints, but for the docstring I would prefer we use `Callable` here, since the web documentation is more user-facing, and it is the defacto standard across the AD python ecosystem. A user would know what `Callable` refers to here, but would be unaware of what `DifferentiableLike` refers to.

Reply:
Thanks, Josh, will do.